src/FactSystem/FactControls/FactCheckBox.qml

```diff
@@ -17,5 +18,5 @@ QGCCheckBox {
 
     text: qsTr("Label")
 
-    onClicked: fact.value = checked ? 1 : 0
+    onClicked: fact.value == checked ? checkedValue : uncheckedValue
```

Comment:
Shit! I've worked around this all over the place in FactCheckBox usage thinking it was come strange Qml quirk! Not a bug in FactCheckBox.